python/ray/_private/metrics_agent.py

```diff
@@ -550,7 +550,13 @@ def record_and_export(self, records: List[Record], global_tags=None):
                 gauge = record.gauge
                 value = record.value
                 tags = record.tags
-                self._record_gauge(gauge, value, {**tags, **global_tags})
+                try:
+                    self._record_gauge(gauge, value, {**tags, **global_tags})
+                except Exception as e:
+                    logger.error(
+                        f"Failed to record metric {gauge.name} with value {value} with tags {tags!r} and global tags {global_tags!r} due to: {e!r}"
+                    )
+                    raise e
```

Comment:
Note that right now, if *any* metric fails to record, all metrics *after* that one don't get recorded - maybe that behavior is wrong? I've kept it as-is here, but if we remove this `raise e` it'll continue on after the bad metric.

Reply:
Yea, I think we should skip the bad one and continue with the good ones. Could you update the PR to remove `raise e`?

Reply:
Done, and added a test!